routers.js

```diff
@@ -0,0 +1,104 @@
+const express = require("express");
+
+const db = require("./data/db.js");
+
+const router = express.Router();
+
+router.get("/", async (req, res) => {
+  try {
+    const posts = await db.find();
+    res.status(200).json(posts);
+  } catch (error) {
+    console.log(error);
+    res
+      .status(500)
+      .json({ message: "The posts information could not be retrieved." });
+  }
+});
+
+router.get("/:id", async (req, res) => {
+  const { id } = req.params;
+  try {
+    const post = await db.findById(id);
+    if (post[0]) {
+      res.status(200).json(post);
+    } else {
+      res
+        .status(404)
+        .json({ message: "The post with the specified ID does not exist." });
+    }
+  } catch (error) {
+    console.log(error);
+    res
+      .status(500)
+      .json({ message: "The post information could not be retrieved." });
+  }
+});
+
+router.post("/", async (req, res) => {
+  const post = req.body;
+  if (!post.title || !post.contents) {
+    res
+      .status(400)
+      .json({ message: "Please provide title and contents for the post." });
+  } else {
+    try {
+      const newPost = await db.insert(post);
+      const findNewPost = await db.findById(newPost.id);
```

Comment:
well done chaining these await calls to sequentially grab what you need! much cleaner than the standard promise syntax with .then().catch() since that gets clunky and overly verbose very quickly